server/pkg/pipeline/pipeline.go

```diff
@@ -1,49 +1,60 @@
-package executor
+package pipeline
 
 import (
 	"context"
 	"fmt"
+	"log"
 	"reflect"
 )
 
-type Pipeline struct {
-	steps  []PipelineStep // N
-	queues []*Queue       // N + 1
+type loggerFn func(string, ...any)
+
+func newLoggerFn(name string) loggerFn {
+	return func(format string, params ...any) {
+		log.Printf("[%s] %s", name, fmt.Sprintf(format, params...))
+	}
+}
+
+type Pipeline[T any] struct {
```

Comment:
I liked this approach of turning the Pipeline generic. This will help us to deal with other queues in the future.